cdap-gateway/src/main/java/co/cask/cdap/gateway/router/RouterPathLookup.java

```diff
@@ -215,9 +217,10 @@ private boolean matches(String[] actual, String... expected) {
   }
 
   private boolean endsWith(String[] actual, String... expectedEnd) {
-    for (int i = expectedEnd.length - 1; i >= 0 && actual.length >= expectedEnd.length; i--) {
+    for (int i = expectedEnd.length - 1; i >= 0; i--) {
       if (!expectedEnd[i].equals("*") &&
```

Comment:
does "a","b" end with "b","*"? According to this, it does.

Reply:
Fixed.